docs/types.rst

```diff
@@ -914,6 +921,54 @@ Of course, you can also directly access the members of the struct without
 assigning it to a local variable, as in
 ``campaigns[campaignID].amount = 0``.
 
+
+Types as input and output parameters
+====================================
+
+.. index:: !function parameters, functions, parameters
+
+Allowed function parameter types
+--------------------------------
+
+**All** :ref:`value-types` variables are allowed as input and output function parameter types.
+
+**All** :ref:`reference-types` variables are allowed as input and output function parameter types for internal functions, and for external functions if you enable the experimental ``ABIEncoderV2`` feature.
+
+.. index:: copying types, referencing types
+
+.. _ref-copy-types:
+
+Copying vs referencing
+----------------------
+
+When you use variables as an input or output parameter to functions, how the
+compiler treats the variable depends on its type.
+
+Any variable that is a :ref:`value-types`, has its value **copied** from the
+output function to the input function.
```

Comment:
"output function" and "input function" should rather be replaced by "calling" and "called" function.

Reply:
@chriseth Good idea, should this also apply to your comment above?

https://github.com/ethereum/solidity/pull/4859#discussion_r214986820